sycl/source/handler.cpp

```diff
@@ -389,6 +389,14 @@ event handler::finalize() {
       // nodes/events of the graph
       if (MEventsWaitWithBarrier.size() == 0) {
         MEventsWaitWithBarrier = GraphImpl->getExitNodesEvents();
+        // Graph-wide barriers take precedence over previous one.
+        // We therefore remove the previous ones from ExtraDependencies list.
+        // The current barrier is then added to this list in the graph_impl.
+        std::vector<detail::EventImplPtr> MEventsBarriers;
+        MEventsBarriers = GraphImpl->removeBarriersFromExtraDependencies();
```

Comment:
We should remove the `M` prefix from this variable name, since it's not a class member variable the prefix is a bit misleading. We could also possibly combine these two lines. 
```cpp
std::vector<detail::EventImplPtr> EventsBarriers = GraphImpl->removeBarriersFromExtraDependencies();
```